qiskit_ibm_runtime/utils/json.py

```diff
@@ -258,9 +252,7 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
         if hasattr(obj, "to_json"):
             return {"__type__": "to_json", "__value__": obj.to_json()}
         if isinstance(obj, QuantumCircuit):
-            kwargs: Dict[str, object] = {"use_symengine": bool(optionals.HAS_SYMENGINE)}
-            if _TERRA_VERSION[0] >= 1:
-                kwargs["version"] = 11
+            kwargs: Dict[str, object] = {"version": min(13, QISKIT_QPY_VERSION)}
```

Comment:
I'd really like to see an to query the service endpoint added to query the max QPY version supported for the `13` here so we can do this automatically. But to avoid confusion on where the 13 comes from how about adding a constant variable to the top:

```python
SERVICE_MAX_SUPPORTED_QPY_VERSION = 13
```

to make it clear that the minimum here is between the max version the service understands, and the most recent QPY version that the installed version of qiskit knows how to generate.

Reply:
done in https://github.com/Qiskit/qiskit-ibm-runtime/pull/2096/commits/1b81b5e6265947c3e36c7278a587a1ee40b02e2d

Reply:
Should this line here also be updated?

```kwargs = {"version": min(SERVICE_MAX_SUPPORTED_QPY_VERSION, QISKIT_QPY_VERSION)}```

Reply:
good catch! done in https://github.com/Qiskit/qiskit-ibm-runtime/pull/2096/commits/d4ff1c8cfc0a1bc943fb91110dc8c1b4b43eb2cd